src/MoBi.Presentation/Tasks/SimulationUpdateTask.cs

```diff
@@ -37,41 +35,53 @@ public class SimulationUpdateTask : ISimulationUpdateTask
+         IEntityPathResolver entityPathResolver,
+         ISimulationFactory simulationFactory,
+         ICloneManagerForBuildingBlock cloneManager,
+         IInteractionTasksForSimulation interactionTasksForSimulation,
+         ISimulationConfigurationFactory simulationConfigurationFactory)
       {
          _context = context;
          _applicationController = applicationController;
          _entityPathResolver = entityPathResolver;
          _simulationFactory = simulationFactory;
+         _cloneManager = cloneManager;
+         _interactionTasksForSimulation = interactionTasksForSimulation;
+         _simulationConfigurationFactory = simulationConfigurationFactory;
       }
 
-      public ICommand UpdateSimulationFrom(IMoBiSimulation simulationToUpdate, IBuildingBlock templateBuildingBlock)
+      public ICommand UpdateSimulation(IMoBiSimulation simulationToUpdate)
       {
-         SimulationConfiguration simulationConfigurationReferencingTemplate;
-         var configurationCommands = new MoBiMacroCommand();
-         var fixedValueQuantities = new PathCache<IQuantity>(_entityPathResolver);
+         var simulationConfiguration = _simulationConfigurationFactory.Create();
+         
+         simulationConfiguration.CopyPropertiesFrom(simulationToUpdate.Configuration);
+         simulationConfiguration.SimulationSettings = _cloneManager.Clone(simulationToUpdate.Configuration.SimulationSettings);
+         
+         simulationToUpdate.Configuration.ModuleConfigurations.Each(moduleConfiguration => { simulationConfiguration.AddModuleConfiguration(templateModuleConfigurationFor(moduleConfiguration)); });
 
-         if (triggersReconfiguration(templateBuildingBlock))
-         {
-            SimulationConfiguration simulationConfiguration;
-            using (var presenter = _applicationController.Start<ICreateSimulationConfigurationPresenter>())
-            {
-               simulationConfiguration = presenter.CreateBasedOn(simulationToUpdate, allowNaming: false);
-            }
+         simulationConfiguration.Individual = templateBuildingBlockFor(simulationToUpdate.Configuration.Individual);
 
-            if (simulationConfiguration == null)
-               return new MoBiEmptyCommand();
+         simulationToUpdate.Configuration.ExpressionProfiles.Each(x => { simulationConfiguration.AddExpressionProfile(templateBuildingBlockFor(x)); });
 
-            simulationConfigurationReferencingTemplate = simulationConfiguration;
-         }
-         else
-         {
-            simulationConfigurationReferencingTemplate = createBuildConfigurationUsingTemplates(simulationToUpdate, templateBuildingBlock);
-            fixedValueQuantities.AddRange(simulationToUpdate.Model.Root.GetAllChildren<IQuantity>(x => x.IsFixedValue));
-         }
+         return updateSimulation(simulationToUpdate, simulationConfiguration, new MoBiMacroCommand(), AppConstants.Captions.UpdatingSimulation);
```

Comment:
waht's this new MacroCommand as a parmaeter? That's very weird. I would expect this to be a collector command (maybe) that you create right before calling the update method and then your return the macrocommand